src/core/server/metrics/integration_tests/server_collector.test.ts

```diff
@@ -79,30 +76,28 @@ describe.skip('ServerMetricsCollector', () => {
 
   it('collect disconnects requests infos', async () => {
     const never = new Promise((resolve) => undefined);
-    const hitSubject = new BehaviorSubject(0);
+    const disconnectRequested$ = new Subject(); // Controls the number of requests in the /disconnect endpoint
+    const disconnectAborted$ = new Subject(); // Controls the abort event in the /disconnect endpoint
 
     router.get({ path: '/', validate: false }, async (ctx, req, res) => {
       return res.ok({ body: '' });
     });
     router.get({ path: '/disconnect', validate: false }, async (ctx, req, res) => {
-      hitSubject.next(hitSubject.value + 1);
-      await never;
+      disconnectRequested$.next();
+      req.events.aborted$.subscribe(() => {
+        disconnectAborted$.next();
+      });
+      await never; // Never resolve the request
       return res.ok({ body: '' });
     });
     await server.start();
 
     await sendGet('/');
-    // superTest.get(path).end needs to be called with a callback to actually send the request.
-    const discoReq1 = sendGet('/disconnect').end(() => null);
-    const discoReq2 = sendGet('/disconnect').end(() => null);
-
-    await hitSubject
-      .pipe(
-        filter((count) => count >= 2),
-        take(1)
-      )
-      .toPromise();
-    await delay(requestWaitDelay); // wait for the requests to send
+    const discoReq1 = sendGet('/disconnect').end();
+    const discoReq2 = sendGet('/disconnect').end();
+
+    // Wait for 2 requests to /disconnect
+    await disconnectRequested$.pipe(take(2)).toPromise();
```

Comment:
NIT: we're subscribing to the observable after having sent the requests. I think the risk of missing an emission is kinda non-existent, but just to be safe, I'd either use a replay effect on `disconnectRequested$`, or:


```ts
const disconnectRequested = disconnectRequested$.pipe(take(2)).toPromise();
const discoReq1 = sendGet('/disconnect').end();
const discoReq2 = sendGet('/disconnect').end();
await disconnectRequested;
```

Reply:
Good catch! I'll make sure to subscribe before triggering any event.